python.d/python_modules/bases/FrameworkServices/SocketService.py

```diff
@@ -16,6 +23,9 @@ def __init__(self, configuration=None, name=None):
         self.unix_socket = None
         self.dgram_socket = False
         self.request = ''
+        self.ssl = False
```

Comment:
Well, are you sure about `ssl` attr? I think it is always true if `self.cert and self.key`.

Reply:
Client certificates aren't inherently mandatory though (think HTTPS for an example of this), and matching on the key and certificate being defined would mandate generating a certificate to use SSL/TLS at all (and some stuff which doesn't require client authentication with certificates will still try to validate the client cert, so requiring a client certificate for such services makes setup _really_ complicated).